cartoframes/data/services/bq_datasets.py

```diff
@@ -162,7 +186,46 @@ def status(self):
     def result(self):
         status = self.status()
 
-        while status != 'done' and status != 'failed':
+        while status not in ('success', 'failure'):
+            time.sleep(1)
+            status = self.status()
+
+        return status
+
+
+class BQUserEnrichmentJob:
```

Comment:
I would extend this one from `BQJob`, or both from an interface one since both are very similar. I know some people would not like it, so it is something to consider, not something to do it

Reply:
Let's see how it evolves. I prefer laziness than adding abstractions prematurely.

Reply:
> I prefer laziness than adding abstractions prematurely

Totally agree. Further, I wouldn't call it laziness but a wise decision: only create abstractions when you're sure they pay off.